packages/models/src/cache/onlineStatus.ts

```diff
@@ -0,0 +1,32 @@
+import { CacheBase } from "./base";
+
+export class OnlineStatus extends CacheBase {
+  readonly key = "online-status";
+
+  async getAll(): Promise<Record<string, string>> {
+    return await this.client.hGetAll(this.key)
+  }
+
+  async addUser(userId: number): Promise<void> {
+    await this.client.hSet(this.key, this.asField(userId), this.encode(1));
+  }
+
+  async removeUser(userId: number): Promise<void> {
+    await this.client.hDel(this.key, this.asField(userId));
+  }
+
+  async isOnline(userId: number): Promise<boolean> {
+    return await this.client.hExists(this.key, this.asField(userId));
+  }
+
+  async isOnlineBatch(userIds: number[]): Promise<boolean[]> {
+    const redisClient = this.client.multi();
+    userIds.forEach(id => redisClient.hExists(this.key, this.asField(id)));
+    const result = await redisClient.exec();
+    return result as unknown as boolean[];
+  }
```

Comment:
It is better to return a map from the user id to its online status `Map<number, boolean>`.  This way the consumers will be able to check if a given user in the batch is online or not by accessing like this `onlineBatchResult[userId]` or `result.get(userId)`.